expyfun/tests/test_experiment_controller.py

```diff
@@ -620,6 +620,65 @@ def test_sound_card_triggering(hide_window):
         ec.load_buffer([1e-2])
         ec.start_stimulus()
         ec.stop()
+    # Test the drift triggers
+    audio_controller.update(SOUND_CARD_DRIFT_TRIGGER=0.001)
+    with ExperimentController(*std_args,
+                              audio_controller=audio_controller,
+                              trigger_controller='sound_card',
+                              n_channels=1,
+                              **std_kwargs) as ec:
+        ec.identify_trial(ttl_id=[1, 0], ec_id='')
+        with pytest.warns(UserWarning, match='Drift triggers overlap with '
+                          'onset triggers.'):
+            ec.load_buffer(np.zeros(ec.stim_fs))
+        ec.start_stimulus()
+        ec.stop()
+    audio_controller.update(SOUND_CARD_DRIFT_TRIGGER=[1.1, 0.3, -0.3,
+                                                      'end'])
+    with ExperimentController(*std_args,
+                              audio_controller=audio_controller,
+                              trigger_controller='sound_card',
+                              n_channels=1,
+                              **std_kwargs) as ec:
+        ec.identify_trial(ttl_id=[1, 0], ec_id='')
+        with pytest.warns(UserWarning, match='Drift trigger at 1.1 seconds '
+                          'occurs outside stimulus window, not stamping '
+                          'trigger.'):
+            ec.load_buffer(np.zeros(ec.stim_fs))
+        ec.start_stimulus()
+        ec.stop()
+    audio_controller.update(SOUND_CARD_DRIFT_TRIGGER=[0.5, 0.501])
+    with ExperimentController(*std_args,
+                              audio_controller=audio_controller,
+                              trigger_controller='sound_card',
+                              n_channels=1,
+                              **std_kwargs) as ec:
+        ec.identify_trial(ttl_id=[1, 0], ec_id='')
+        with pytest.warns(UserWarning, match='Some 2-triggers overlap, times '
+                          'should be at least 0.02 seconds apart.'):
```

Comment:
This really depends on what the trigger length is set to be. But maybe this is fins a general suggestion, as there is no real reason why you might want the 2's to be any closer than that.

Reply:
Right, just realized in `_sound_controller` I have it use `trig2time` for the suggestion, but I copied the actually error output for the test. Is that ok, or should I figure to make that adapt to the trigger length as well?

Reply:
`match` is a regex and we just want to make sure it emits some warning so I would just do `match='Some 2-triggers overlap.*'`

Reply:
Don't know how to reference a different part of the code in a reply, but in `_sound_controller.py` I have